File: app/src/main/java/collection/ex03.java
package collection;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

public class ex03 {
    static class Student {
        int id;
        String name;
        int score;

        Student(int id, String name, int score) {
            this.id = id;
            this.name = name;
            this.score = score;
        }

        @Override
        public String toString() {
            return "Student{id=" + id + ", name='" + name + "', score=" + score + "}";
        }
    }

    public static void main(String[] args) {
        List<Student> students = new ArrayList<>();
        students.add(new Student(3, "Kim", 90));
        students.add(new Student(1, "Lee", 95));
        students.add(new Student(2, "Park", 95));

        students.sort(
                Comparator.comparingInt((Student s) -> s.score).reversed()
                        .thenComparingInt(s -> s.id)
        );

        students.forEach(System.out::println);
    }
}

File: app-kotlin/src/main/kotlin/collection/ex03.kt
package collection

class ex03 {
    data class Student(val id: Int, val name: String, val score: Int)

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val students = mutableListOf(
                Student(3, "Kim", 90),
                Student(1, "Lee", 95),
                Student(2, "Park", 95)
            )

            val sorted = students.sortedWith(compareByDescending<Student> { it.score }.thenBy { it.id })
            sorted.forEach { println(it) }
        }
    }
}
